mlflow/store/tracking/dbmodels/models.py

```diff
@@ -463,3 +465,161 @@ def to_mlflow_entity(self):
         :return: :py:class:`mlflow.entities.Param`.
         """
         return Param(key=self.key, value=self.value)
+
+
+class SqlDataset(Base):
+    __tablename__ = "datasets"
+    __table_args__ = (
+        PrimaryKeyConstraint("experiment_id", "name", "digest", name="dataset_pk"),
+        Index(f"index_{__tablename__}_dataset_uuid", "dataset_uuid"),
+        Index(
+            f"index_{__tablename__}_experiment_id_dataset_source_type",
+            "experiment_id",
+            "dataset_source_type",
+        ),
+    )
+
+    dataset_uuid = Column(String(36), nullable=False)
+    """
+    Dataset UUID: `String` (limit 36 characters). Defined as *Non-null* in schema.
+    Part of *Primary Key* for ``datasets`` table.
+    """
+    experiment_id = Column(Integer, ForeignKey("experiments.experiment_id"))
+    """
+    Experiment ID to which this dataset belongs: *Foreign Key* into ``experiments`` table.
+    """
+    name = Column(String(500), nullable=False)
+    """
+    Param name: `String` (limit 500 characters). Defined as *Non-null* in schema.
+    Part of *Primary Key* for ``datasets`` table.
+    """
+    digest = Column(String(36), nullable=False)
+    """
+    Param digest: `String` (limit 500 characters). Defined as *Non-null* in schema.
+    Part of *Primary Key* for ``datasets`` table.
+    """
+    dataset_source_type = Column(String(36), nullable=False)
+    """
+    Param dataset_source_type: `String` (limit 36 characters). Defined as *Non-null* in schema.
+    """
+    dataset_source = Column(String(5000), nullable=False)
+    """
+    Param dataset_source: `String` (limit 5000 characters). Defined as *Non-null* in schema.
+    """
+    dataset_schema = Column(String(5000), nullable=True)
+    """
+    Param dataset_schema: `String` (limit 5000 characters).
+    """
+    dataset_profile = Column(String(5000), nullable=True)
```

Comment:
5000 is a fair bit smaller than MySQL TEXT and MEDIUMTEXT columns. Can we use those sizes, or is there a limitation preventing that?

Reply:
I'm not sure if there is a limitation. For `Tag` and `ExperimentTag`, we use this same 5000 limit despite having those as `text` on the databricks side.

Reply:
@dbczumar looks like we can use `TEXT`. https://stackoverflow.com/questions/47644739/what-column-type-does-sqlalchemy-use-for-text-on-mysql

I don't think we should use `MEDIUMTEXT` since that is a mysql specific construct that may not exist in other implementations. https://docs.sqlalchemy.org/en/20/dialects/mysql.html#sqlalchemy.dialects.mysql.MEDIUMTEXT